iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -183,7 +184,11 @@ public static boolean dropTable(Configuration conf, Properties props) {
    * @return true if the Catalog is HiveCatalog
    */
   public static boolean hiveCatalog(Configuration conf, Properties props) {
-    String catalogName = props.getProperty(InputFormatConfig.CATALOG_NAME);
+    return hiveCatalog(conf, props::getProperty);
+  }
+
+  public static boolean hiveCatalog(Configuration conf, UnaryOperator<String> props) {
```

Comment:
Why is this necessary?

Reply:
because i need that method signature

Reply:
we can drop it if the tests pass without this check: https://github.com/apache/hive/pull/5498/commits/f71c124d85efe91431f93f9d2e79e7f46f08338f

Reply:
few failed, I've fixed them and removed this method